configure.ac

```diff
@@ -24,16 +24,18 @@ AC_ENABLE_SHARED
 AC_ENABLE_STATIC
 AC_PROG_CC
 AM_PROG_CC_C_O
-AS_IF([AS_CASE([$host_os],[openedition*],  [false], [true])], [
-  CC_CHECK_CFLAGS_APPEND([-pedantic])
-])
 CC_FLAG_VISIBILITY #[-fvisibility=hidden]
 CC_CHECK_CFLAGS_APPEND([-g])
-CC_CHECK_CFLAGS_APPEND([-std=gnu89])
-CC_CHECK_CFLAGS_APPEND([-Wall])
-CC_CHECK_CFLAGS_APPEND([-Wextra])
-CC_CHECK_CFLAGS_APPEND([-Wno-unused-parameter])
-CC_CHECK_CFLAGS_APPEND([-Wstrict-prototypes])
+AS_IF([test "x$GCC" = xyes], [
+  AS_IF([AS_CASE([$host_os], [openedition*], [false], [true])], [
+    CC_CHECK_CFLAGS_APPEND([-pedantic])
+  ])
+  CC_CHECK_CFLAGS_APPEND([-std=gnu89])
+  CC_CHECK_CFLAGS_APPEND([-Wall])
+  CC_CHECK_CFLAGS_APPEND([-Wextra])
+  CC_CHECK_CFLAGS_APPEND([-Wno-unused-parameter])
+  CC_CHECK_CFLAGS_APPEND([-Wstrict-prototypes])
+])
```

Comment:
Yes, but it doesn't do a good job of detecting whether failure occurs. This is what it detects:
```
checking if cc supports -fvisibility=hidden... yes
checking if cc supports -g flag... yes
checking if cc supports -std=gnu89 flag... yes
checking if cc supports -Wall flag... no
checking if cc supports -Wextra flag... no
checking if cc supports -Wno-unused-parameter flag... no
checking if cc supports -Wstrict-prototypes flag... no
```

That is broken, since neither `-fvisibility=hidden` or `-std=gnu89` are actually supported flags. What is broken is that Studio will pass unknown flags to the linker, in case the flags are destined for the linker, but since `autoconf` is not linking, it does not blow up until during the actual build. From `config.log`:

```
configure:4286: checking if cc supports -fvisibility=hidden
configure:4303: cc -c -g -pedantic -errwarn=%all -fvisibility=hidden  conftest.c >&5
cc: Warning: Option -fvisibility=hidden passed to ld, if ld is invoked, ignored otherwise
configure:4303: $? = 0
configure:4322: result: yes
```

Similarly, in the `-std=gnu89` case, the `-st` is interpreted as an actual compiler option, and the remaining `-d=gnu89` is treated as a linker flag:
```
configure:4379: checking if cc supports -std=gnu89 flag
configure:4394: cc -c -g -pedantic -g -std=gnu89  conftest.c >&5
cc: Warning: Option -d=gnu89 passed to ld, if ld is invoked, ignored otherwise
configure:4394: $? = 0
configure:4411: result: yes
```

That then errors the first time the real build tries to link:
```
$ cc -std=gnu89 -o test test.c
cc: Warning: Option -d=gnu89 passed to ld, if ld is invoked, ignored otherwise
ld: fatal: option '-d' has illegal argument '=gnu89'
```

What needs to happen here is that Oracle needs to add a flag (or tell us about some hidden flag) that makes unknown compiler options a fatal error so that `autoconf` does its job correctly, or `autoconf` needs to be smarter and test a full link when auto-detecting the options.

Since we know these are GCC-isms, I figured it is fine to scope them to GCC detection.

Reply:
Okay, sounds reasonable. Thanks for the explanation.